src/TimeTrackerWindow.cpp

```diff
@@ -119,19 +122,42 @@ TimeTrackerWindow::MessageReceived(BMessage* message)
 	}
 	case MENU_TASK_DELETE:
 	{
+		
 		int32 selection = m_ListView->CurrentSelection();
 		if (selection >= 0) {
-			BListItem* temp = m_ListView->RemoveItem(selection);
-			delete(temp);
+			BAlert* alert  = new BAlert("Confirm delete", 
+						"Are you sure you want to delete the selected task(s)?", 
+					NULL, "OK", "Cancel", B_WIDTH_AS_USUAL, B_WARNING_ALERT);
+		
+			alert->SetShortcut(1, B_ESCAPE);
+			int32 button_inndex = alert->Go();
+		
+			if (button_inndex == 0) {
+				int32 s = 0;
+				do	
+				{	
+					if (m_ListView->IsItemSelected(s))
+						delete	m_ListView->RemoveItem(s);
+				}
+				while((s = m_ListView->CurrentSelection())>=0);
+			}
 		}
 		break;
 	}
 	case MENU_TASK_RESET:
 	{
 		int32 selection = m_ListView->CurrentSelection();
 		if (selection >= 0) {
-			TaskListItem* temp = (TaskListItem*)m_ListView->ItemAt(selection);
-			temp->ResetTime();
+			int32 s = 0;
+			do	
+			{	
+				if (m_ListView->IsItemSelected(s)) {
+					TaskListItem*	temp = (TaskListItem*)m_ListView->ItemAt(s);
+					temp->ResetTime();
+					m_ListView->Deselect(s);
+				}			
+			}
+			while((s = m_ListView->CurrentSelection())>=0);
```

Comment:
Follow Haiku coding style: while goes on previous line; { for do goes on previous line.